json-endpoints/class.wpcom-json-api-update-comment-endpoint.php

```diff
@@ -102,6 +102,17 @@ function new_comment( $path, $blog_id, $post_id, $comment_parent_id ) {
 				if ( !isset( $user->ID ) ) {
 					$user->ID = 0;
 				}
+
+				// If we have a user with an external ID saved, we can use it.
+				if (
+					! $auth_required
+					&& $user->ID
+					&& (
+						$author = get_user_by( 'id', intval( $user->ID ) )
```

Comment:
I thought that external user id was saved in `wpcom_user_id` user meta, and it looks like we are trying to match that here against the local `id` instead? I'm probably missing something, but I'm not sure what. :)

Reply:
The only place that sets `wpcom_user_id`, that I know of, is SSO. So, it would probably only be available if the user has SSO'd at least once.